contracts/AirDrop.sol

```diff
@@ -0,0 +1,242 @@
+        _owner = msg.sender;
+    }
+
+    /**
+     * @dev send ERC20 tokens from 'owner_' balance to SC
+     *
+     * @param amount_ - how many tokens deposits
+     *
+     * NOTE : this function use wrapper of safeERC20 library
+     * check 'SafeERC20.sol' for more information
+     */
+    function depositTokens(uint256 amount_) external override onlyOwner {
+        require(amount_ != 0, "Error : 'Amount' , equal to 0");
+        token.safeTransferFrom(msg.sender, address(this), amount_);
+        emit DepositTokens(amount_);
+    }
+
+    /**
+     * @dev send ethers from 'owner_' balance to SC
+     *
+     * NOTE : if you send ether to 0 address , ethers will be blocked.
+     */
+    function depositEther() external payable override onlyOwner {
+        require(msg.value != 0, "Error : 'Amount' , equal to 0");
+        uint256 value = uint256(msg.value);
+        emit DepositEther(value);
+    }
+
+    /**
+     * @dev make tokens available for members to withdraw
+     *
+     * @param recepients_ - address of member
+     * @param amount_ - how many tokens will be available
+     * @param deadline_ - how long signature will be valid
+     * @param v_ - can be only 27 or 28
+     * @param r_ - contain info to recover signer
+     * @param s_ - contain info to recover signer
+     *
+     * NOTE : more information 'EIP712'
+     */
+    function dropTokens(
+        address recepients_,
+        uint256 amount_,
+        uint256 deadline_,
+        uint8 v_,
+        bytes32 r_,
+        bytes32 s_
+    ) external override onlyOwner {
+        require(
+            amount_ != 0 && recepients_ != address(0) && deadline_ != 0,
+            "Error :'recepients_' or 'amount_' or 'deadline_' equal to 0"
+        );
+        bytes32 structHash = keccak256(
+            abi.encode(DROP_HASH, amount_, deadline_, recepients_)
+        );
+        bytes32 digest = EIP712._hashTypedDataV4(structHash);
+
+        address msgSigner = ECDSA.recover(digest, v_, r_, s_);
+        require(_owner == msgSigner, "invalid signature");
+        require(
+            block.timestamp < deadline_,
+            "Error : signed transaction expired"
+        );
+
+        balanceOfTokens[recepients_] += amount_;
+        token.approve(recepients_, amount_);
+        emit DropTokens(recepients_, amount_, deadline_);
+    }
+
+    /**
+     * @dev make ethers available for members to withdraw
+     *
+     * @param recepients_ - address of member
+     * @param amount_ - how many ethers will be available
+     * @param deadline_ - how long signature will be valid
+     * @param v_ - can be only 27 or 28
+     * @param r_ - contain info to recover signer
+     * @param s_ - contain info to recover signer
+     *
+     * NOTE : more information 'EIP712'
+     */
+    function dropEther(
+        address recepients_,
+        uint256 amount_,
+        uint256 deadline_,
+        uint8 v_,
+        bytes32 r_,
+        bytes32 s_
+    ) external override onlyOwner {
```

Comment:
You have a Signature Replay security issue

Reply:
first: owner can use the same signature many times (read about nonce in signature) 
second: owner can use the same signature for both dropTokens and dropEthers function. let's make it impossible. Advise: add flag "bool isEther " to Drop container as example